test/functional/blockprod_generate_pos_blocks.py

```diff
@@ -0,0 +1,372 @@
+
+        return (new_vrf_private_key, new_vrf_public_key)
+
+    def run_test(self):
+        #
+        # Transfer Genesis UTXO to AnyoneCanSpend
+        #
+
+        block_input_data = block_input_data_obj.encode({
+            "PoS": {
+                "stake_private_key": self.stake_private_key(GENESIS_STAKE_PRIVATE_KEY),
+                "vrf_private_key": self.vrf_private_key(GENESIS_VRF_PRIVATE_KEY),
+                "pool_id": self.genesis_pool_id(),
+                "kernel_inputs": [
+                        {
+                            "Utxo": {
+                                "id": {
+                                    "BlockReward": self.previous_block_id()
+                                },
+                                "index": 1,
+                            },
+                        },
+                ],
+                 "kernel_input_utxo": [
+                    {
+                        "CreateStakePool": [
+                            self.genesis_pool_id(),
+                            {
+                                "value": "4000000000000000",
+                                "staker": {
+                                    "PublicKey": self.stake_public_key(GENESIS_STAKE_PUBLIC_KEY),
+                                },
+                                "vrf_public_key": self.vrf_public_key(GENESIS_VRF_PUBLIC_KEY),
+                                "decommission_key": {
+                                    "PublicKey": self.stake_public_key(GENESIS_STAKE_PUBLIC_KEY),
+                                },
+                                "margin_ratio_per_thousand": 1000,
+                                "cost_per_block" : "0"
+                            },
+                        ],
+                    }
+                ],
+            }
+        }).to_hex()[2:]
+
+        transfer_transaction = {
+            "transaction": {
+                "version": 1,
+                "flags": 0,
+                "inputs": [
+                    {
+                        "Utxo": {
+                            "id": {
+                                "BlockReward": self.previous_block_id()
+                            },
+                            "index": 0,
+                        },
+                    }
+                ],
+                "outputs": [
+                    {
+                        "Transfer": [
+                            {
+                                "Coin": "10000000000000000",
+                            },
+                            "AnyoneCanSpend",
+                        ],
+                    },
+                ],
+            },
+            "signatures": [
+                {
+                    "NoSignature": None,
+                },
+            ],
+        }
+
+        (transfer_transaction_encoded, transfer_transaction_id) = self.pack_transaction(transfer_transaction)
+        self.generate_block(block_input_data, [transfer_transaction_encoded])
```

Comment:
Please split this test into two tests. In one you only use the genesis pool, and in another, you use the one you created. Each prove different properties.

Keep in mind that the one with the pool may not work in the future if we enforce having stake balances come from sealed blocks.

Reply:
To be clearer, the shortest path here is to add another functional test where you just stake from the genesis pool.